src/versions/drafts/replaceWithDraftIfAvailable.ts

```diff
@@ -40,9 +41,12 @@ const replaceWithDraftIfAvailable = async <T extends TypeWithID>({
   };
 
   if (entityType === 'collection') {
+    const customIDField = entity.fields.find((field) => fieldAffectsData(field) && field.name === 'id');
+    const parentID = customIDField ? doc.id : new mongoose.Types.ObjectId(doc.id);
+
     queryToBuild.where.and.push({
       parent: {
-        equals: doc.id,
+        equals: parentID,
```

Comment:
I think the old parent fields are still string fields? Will this brick when upgraded from an older version?

Or will this be later parsed to an in: [string, objectId] statement?